src/extension/providers/commonProvider/index.ts

```diff
@@ -0,0 +1,43 @@
+import { EventEmitter } from "events";
+import { postMessage } from "~/extension/providers/postMessage";
+import { PromiseQueue } from "~/extension/providers/promiseQueue";
+
+export default class CommonProvider {
+  enabled: boolean;
+  private _eventEmitter: EventEmitter;
+  private _queue: PromiseQueue;
+
+  constructor() {
+    this.enabled = false;
+    this._eventEmitter = new EventEmitter();
+    this._queue = new PromiseQueue();
+  }
+
+  on(...args: Parameters<EventEmitter["on"]>) {
+    if (!this.enabled) {
```

Comment:
Could this enabled check be moved to the base class, and just pass the method of the name being called? something like `this._checkEnabled("on");`